### PR TITLE
Save User Model in SharedPref

### PR DESCRIPTION
**Pull request by SKrPl**
**Issue**: https://github.com/openMF/android-client/issues/339

saves User model in shared preferences using Gson. Shows username of logged in User instead of Client.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/login/LoginActivity.java [hunk 1]
```diff
@@ -158,10 +158,8 @@ public boolean validateUserInputs() {
     public void onLoginSuccessful(User user) {
         Toaster.show(findViewById(android.R.id.content), getString(R.string.toast_welcome) + " "
                 + user.getUsername());
-        // Saving userID
-        PrefManager.setUserId(user.getUserId());
-        // Saving user's token
-        PrefManager.saveToken("Basic " + user.getBase64EncodedAuthenticationKey());
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/login/LoginActivity.java]: Do not remove these, Token is being used, whenever we are making any request after login.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 2]
```diff
@@ -200,69 +188,19 @@ public void run() {
     //TODO Setup In MVP
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: remove unnecessary comment.

File: mifosng-android/src/main/java/com/mifos/utils/PrefManager.java [hunk 3]
```diff
@@ -85,6 +88,16 @@ public static void putStringSet(String preferencesKey, Set<String> values) {
      * Authentication
      */
 
+    public static void saveUser(User user) {
+        Gson gson = new Gson();
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/utils/PrefManager.java]: make Gson Object global.

File: mifosng-android/src/main/java/com/mifos/utils/PrefManager.java [hunk 4]
```diff
@@ -21,6 +23,7 @@
     private static final String INSTANCE_DOMAIN = "preferences_domain";
     private static final String PORT = "preferences_port";
     private static final String USER_STATUS = "user_status";
+    private static final String USER_SAVE = "user_save";
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/utils/PrefManager.java]: ambiguous naming, Change to USER_DETAILS

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/login/LoginActivity.java (hunk 1 above)
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/login/LoginActivity.java]: and UserId too.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java (hunk 2 above)
Comment (SKrPl) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: should i remove the TODO MVP comment ?

**Comment by therajanmaurya** [comment]
One more thing, do not open a PR with your fork master. Please make new branch, according to feature and open again PR.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java (hunk 2 above)
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: Yes

File: mifosng-android/src/main/java/com/mifos/utils/PrefManager.java [hunk 5]
```diff
@@ -21,6 +23,9 @@
     private static final String INSTANCE_DOMAIN = "preferences_domain";
     private static final String PORT = "preferences_port";
     private static final String USER_STATUS = "user_status";
+    private static final String USER_DETAILS = "user_details";
+
+    private static final Gson GSON = new Gson();
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/utils/PrefManager.java]: change to  private static Gson gson = new Gson();

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 6]
```diff
@@ -194,75 +182,24 @@ public void run() {
...
+        TextView textViewUsername = (TextView) mNavigationHeader.findViewById(R.id.tv_user_name);
+        textViewUsername.setText(loggedInUser.getUsername());
+
+        // no profile picture credential, using dummy profile picture
+        ImageView imageViewUserPicture = (ImageView) mNavigationHeader
+                .findViewById(R.id.iv_user_picture);
+        Glide.with(getApplicationContext())
+                .load("")
+                .placeholder(R.drawable.ic_account_circle)
+                .into(imageViewUserPicture);
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: remove this glide code and set a default image in this imageView

**Comment by SKrPl** [comment]
pull request from a different branch with changes .... please check ... https://github.com/openMF/android-client/pull/343

**Comment by therajanmaurya** [comment]
@SKrPl Please close this PR. this no more needed.

**Pull request closed without merging**